retro_data_structures/properties/base_property.py

```diff
@@ -35,6 +43,36 @@ def from_json(cls: typing.Type[Self], data: typing.Any) -> Self:
 
     def to_json(self) -> typing.Any:
         raise NotImplementedError()
+    
+    def _is_property_mrea_or_mlvl(self, field: dataclasses.Field) -> bool:
+        return (('asset_types' in field.metadata)
+            and (("MLVL" in field.metadata['asset_types'])
+                or ("MREA" in field.metadata['asset_types'])
+            ))
+
+    def _dependencies_for_field(self, field: dataclasses.Field, asset_manager: AssetManager, is_mlvl: bool = False) -> typing.Iterator[Dependency]:
+        if issubclass(field.type, BaseProperty):
+            if is_mlvl and field.type.__name__ == "AreaAttributes":
```

Comment:
Couldn't this code be in `AreaAttributes`?

Reply:
it's such a pain to do that for a templated type... I could get away with it for AnimationParameters since we already generated that whole file manually but AreaAttributes comes from a real template

Reply:
Worst thing is the `is_mlvl` check, hm. Could have a flag for types which makes we generate an empty `dependencies_for`, but ugh.

Could at least just move this `if` to `dependencies_for` though. Fits better there.